sam.c

```diff
@@ -421,6 +448,7 @@ int bam_read1(BGZF *fp, bam1_t *b)
         bgzf_read(fp, b->data + c->l_qname, b->l_data - c->l_qname) != b->l_data - c->l_qname)
         return -4;
     if (fp->is_be) swap_data(c, b->l_data, b->data, 0);
+    bam_tag2cigar(b);
```

Comment:
Given the memory allocation, please make bam_tag2cigar return int and check the value here.